copy/copy.go

```diff
@@ -469,12 +490,19 @@ func (ic *imageCopier) copyLayers(ctx context.Context) error {
 
 	progressBars := make([]*pb.ProgressBar, numLayers)
 	for i, srcInfo := range srcInfos {
-		bar := createProgressBar(srcInfo, "blob", nil)
+		bar := ic.c.createProgressBar(srcInfo, "blob")
 		progressBars[i] = bar
 	}
 
 	progressPool := pb.NewPool(progressBars...)
 	progressPool.Output = ic.c.reportWriter
+
+	// reset the output if it's a file but not a tty to avoid long and
+	// hard to parse indeterministic output from the progress bars
+	if !isFileButNoTTY(progressPool.Output) {
```

Comment:
(Non-blocking: It would be a tiny bit more readable for this to look the same as the condition in `createProgressBar` (i.e. probably for both to use the same `if`/`else`.)

Reply:
I like your proposal. The print decision can be based on `== ioutil.Discard` or we default `.progressBarOutput` to nil and set it there to `iotul.Discard` (maybe faster).